### PR TITLE
chore: add api versioning to registration

### PR DESCRIPTION
**Pull request by Abhinandan-Purkait**
Signed-off-by: Abhinandan Purkait <abhinandan.purkait@mayadata.io>

### CODE REVIEW

**Review by Ab-hishek — APPROVED** [review]

File: protobuf/v1/registration.proto [hunk 1]
```diff
@@ -13,6 +13,14 @@ message RegisterRequest {
   string grpc_endpoint = 2;
   // uuid generated by the instance on startup
   google.protobuf.StringValue instance_uuid = 3;
+  // latest api version supported by the dataplane
+  APIVersion api_version = 4;
```
Comment (tiagolobocastro) [inline protobuf/v1/registration.proto:17]: what about adding another message which has the version for now, and in the future we can add other things like capabilities? Or is it better to keep it separate?

File: protobuf/v1/registration.proto [hunk 2]
```diff
@@ -13,6 +13,14 @@ message RegisterRequest {
...
+}
+
+// latest api version supported by the dataplane
+enum APIVersion {
```
Comment (tiagolobocastro) [inline protobuf/v1/registration.proto:21]: Would it be worth reporting a list of versions?

Comment (Abhinandan-Purkait) [reply protobuf/v1/registration.proto:21]: Done.

File: protobuf/v1/registration.proto (hunk 1 above)
Comment (Abhinandan-Purkait) [reply protobuf/v1/registration.proto]: Capabilites now will be a part of host, MayaStorFeatures as per v1 api.

**Review by tiagolobocastro — APPROVED** [review]

**Pull request merged**